pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -569,7 +585,7 @@ private InstancePartitions getInstancePartitions(TableConfig tableConfig,
         LOGGER.info("Removing instance partitions: {} from ZK if it exists", instancePartitionsName);
         InstancePartitionsUtils.removeInstancePartitions(_helixManager.getHelixPropertyStore(), instancePartitionsName);
       }
-      return instancePartitions;
+      return Pair.of(instancePartitions, true);
```

Comment:
Here we want to check if there is existing instance partitions, and return false if there is no existing instance partitions

Reply:
So just to double check, if there are existing instance partitions then it should return NO_OP?